src/solver/simulation/shave-peaks-by-remix-hydro.cpp

```diff
@@ -235,24 +248,33 @@ RemixHydroOutput shavePeaksByRemixingHydro(const std::vector<double>& DispatchGe
                 {
                     break;
                 }
-
-                std::vector<double> intermediate_level(levels.begin()
-                                                         + std::min(hourBottom, hourPeak),
-                                                       levels.begin()
-                                                         + std::max(hourBottom, hourPeak));
                 double max_pic, max_creux;
-                if (hourBottom < hourPeak)
+
+                if (reservoirManagement)
                 {
-                    max_pic = capa;
-                    max_creux = *std::min_element(intermediate_level.begin(),
-                                                  intermediate_level.end());
+                    std::vector<double> intermediate_level(levels.begin()
+                                                             + std::min(hourBottom, hourPeak),
+                                                           levels.begin()
+                                                             + std::max(hourBottom, hourPeak));
```

Comment:
```suggestion
                    auto intermediate_level = levels | std::views::drop(std::min(hourBottom, hourPeak)) 
                                  | std::views::take(std::abs(hourBottom - hourPeak));
```

Drop [begin, min(hourBottom, hour peak] then take abs(hourPeak - hourBottom);
If hourBottom < hourPeak then take hourPeak - hourBottom element starting hourBottom
If hourPeak < hourBottom them take hourBottom - hourPeak element starting hourPeak

Reply:
I find the original code easier to read, and chaining filters isn't supported by all compilers (e.g gcc 10) as @payetvin noticed

Reply:
Works fine with gcc10 (in this case) https://godbolt.org/z/4aneqYeKP

I personally find it clearer that pointer arithmetic tbh

Reply:
Another point. I don't know the size of intermediate_level and how much time we pass in the function, but using vector(it, it) creates a vector with copy of the values, whereas using filter views operates on the collection.
Performance-wise it may be more interesting to use views

Reply:
I used `std::span`, which avoids copies and is readable